src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -141,7 +141,13 @@ exprt value_set_dereferencet::dereference(
     }
   }
 
-  // type of the object
+  return handle_dereference_base_case(pointer, display_points_to_sets);
+}
+
+exprt value_set_dereferencet::handle_dereference_base_case(
```

Comment:
@thomasspriggs The bodies of the branches could also be factored out, I just didn't bother for now since they’re short. The reason why I factored this out was to separate the decision whether or not to recurse from the handling of the case where we don’t recurse, because these are different... functions.